webapp/packages/core-resource/src/Resource/Resource.ts

```diff
@@ -107,6 +107,35 @@ export abstract class Resource<
     return ResourceKeyUtils.isIntersect(key, nextKey, this.isKeyEqual);
   }
 
+  /**
+   * Checks
+   * @param param - Resource key
+   * @param second - Resource key
+   * @returns {boolean} Returns true if key can is the same by all key-values
+   */
+  isEqual(param: ResourceKey<TKey>, second: ResourceKey<TKey>): boolean {
+    if (param === second) {
+      return true;
+    }
+
+    if (isResourceAlias(param) && isResourceAlias(second)) {
+      param = this.aliases.transformToAlias(param);
+      second = this.aliases.transformToAlias(second);
+
+      return param.isEqual(second);
+    }
+
+    if (isResourceAlias(param) || isResourceAlias(second)) {
+      return false;
+    }
+
+    if (isResourceKeyList(param) && isResourceKeyList(second)) {
+      return isArraysEqual(param, second);
+    }
```

Comment:
this will work only for key list of primitives and not objects, we need to add `isEqual` to `ResourceKeyList`